### PR TITLE
Утшо Момен - Лабораторная работа #1

### PR DESCRIPTION
**Pull request by utshomomen**

### CODE REVIEW

File: modules/complex-number/test/test_utsho_momen_complex_number.cpp [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+#include <gtest/gtest.h> 
+
+#include "include/complex_number.h" 
+
+TEST(Utsho_Momen_ComplexNumberTest, Init_Imaginary_To_Zero) {
+	// Arrange & Act
+	ComplexNumber z;
+
+	// Assert
+	ASSERT_NO_THROW(0, z.im());
```
Comment (alexbatashev) [inline modules/complex-number/test/test_utsho_momen_complex_number.cpp]:
```suggestion
	EXPECT_FLOAT_EQ(0, z.getIm());
```

File: modules/complex-number/test/test_utsho_momen_complex_number.cpp [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+}
+
+TEST(Utsho_Momen_ComplexNumberTest, Set_Init_Values) {
+	// Arrange & Act
+	ComplexNumber z1 = ComplexNumber(5, -2);
+	ComplexNumber z2 = ComplexNumber(1, -1);
+
+    // Assert
+	ASSERT_NO_THROW(5, z1.re());
```
Comment (alexbatashev) [inline modules/complex-number/test/test_utsho_momen_complex_number.cpp]:
```suggestion
	EXPECT_FLOAT_EQ(5, z1.getRe());
```

**Comment by lumpofsugar** [comment]
👍

**Comment by utshomomen** [comment]
@lumpofsugar  @ArtemSadikov

**Comment by lumpofsugar** [comment]
👍

**Comment by IsinZ** [comment]
👍

**Comment by utshomomen** [comment]
> Please, refer to [Google Test docs](https://github.com/google/googletest/blob/master/googletest/docs/primer.md) and [original tests](https://github.com/UNN-ITMM-Software/devtools-course-practice/blob/master/modules/complex-number/test/test_complex_number.cpp)

я исправил ошибку

**Comment by ArtemSadikov** [comment]
👍

File: modules/complex-number/test/test_utsho_momen_complex_number.cpp [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+TEST(Utsho_Momen_ComplexNumberTest, Init_Imaginary_To_Zero) {
+    // Arrange & Act
+    ComplexNumber z;
+
+    // Assert
+    EXPECT_FLOAT_EQ(0, z.getIm());
+}
+
+TEST(Utsho_Momen_ComplexNumberTest, Divide_By_Zero) {
```
Comment (alexbatashev) [inline modules/complex-number/test/test_utsho_momen_complex_number.cpp]: Duplicate https://github.com/UNN-ITMM-Software/devtools-course-practice/blob/master/modules/complex-number/test/test_complex_number.cpp#L114

File: modules/complex-number/test/test_utsho_momen_complex_number.cpp [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    // Arrange
+    ComplexNumber z1(45.65, 13.35);
+    ComplexNumber z2(0.0, 34.75);
+
+    // Act & Assert
+    EXPECT_NO_THROW(z1 / z2);
+}
+
+TEST(Utsho_Momen_ComplexNumberTest, Comparison_Equal) {
```
Comment (alexbatashev) [inline modules/complex-number/test/test_utsho_momen_complex_number.cpp]: Duplicate https://github.com/UNN-ITMM-Software/devtools-course-practice/blob/master/modules/complex-number/test/test_complex_number.cpp#L131

File: modules/complex-number/test/test_utsho_momen_complex_number.cpp (hunk 3 above)
Comment (utshomomen) [reply modules/complex-number/test/test_utsho_momen_complex_number.cpp]: Solved the problems.

File: modules/complex-number/test/test_utsho_momen_complex_number.cpp (hunk 4 above)
Comment (utshomomen) [reply modules/complex-number/test/test_utsho_momen_complex_number.cpp]: Solved the problems.

File: modules/complex-number/test/test_utsho_momen_complex_number.cpp [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+TEST(Utsho_Momen_ComplexNumberTest, Init_Imaginary_To_Zero) {
+    // Arrange & Act
+    ComplexNumber z;
+
+    // Assert
+    EXPECT_FLOAT_EQ(0, z.getIm());
+}
+
+TEST(Utsho_Momen_ComplexNumberTest, Equal_To_Zero) {
```
Comment (alexbatashev) [inline modules/complex-number/test/test_utsho_momen_complex_number.cpp]: How do you mean?
This is simple multiplication

File: modules/complex-number/test/test_utsho_momen_complex_number.cpp (hunk 3 above)
Comment (utshomomen) [reply modules/complex-number/test/test_utsho_momen_complex_number.cpp]: > Duplicate https://github.com/UNN-ITMM-Software/devtools-course-practice/blob/master/modules/complex-number/test/test_complex_number.cpp#L114

Solved.

File: modules/complex-number/test/test_utsho_momen_complex_number.cpp (hunk 4 above)
Comment (utshomomen) [reply modules/complex-number/test/test_utsho_momen_complex_number.cpp]: > Duplicate https://github.com/UNN-ITMM-Software/devtools-course-practice/blob/master/modules/complex-number/test/test_complex_number.cpp#L131

Solved the problems.

File: modules/complex-number/test/test_utsho_momen_complex_number.cpp (hunk 5 above)
Comment (utshomomen) [reply modules/complex-number/test/test_utsho_momen_complex_number.cpp]: I need to write in details i think.

Comment (utshomomen) [reply modules/complex-number/test/test_utsho_momen_complex_number.cpp]: > How do you mean?
> This is simple multiplication

Already rewrite the following mistakes.Thanks in advance.

**Comment by lumpofsugar** [comment]
💩

**Comment by AVBalynin** [comment]
👳🏾‍♀️

**Comment by lumpofsugar** [comment]
💣💣💣

**Comment by utshomomen** [comment]
@alexbatashev Everything is solved i hope.

**Comment by utshomomen** [comment]
> 💣💣💣

**Pull request merged**